### PR TITLE
Configure date sort; add contextual alert.

### LINKED ISSUE #1470: Provide context for users on "sort by" functionality
As a DLME user, I sometimes am confused by the sort behavior. Ideally the UI would provide enough context for me to interpret the sort results.

In DLME the sort by functionality relies on data quality/completeness. We are planning on adding sort by date to the existing 'relevance', 'title' and 'creator' options via https://github.com/sul-dlss/dlme/issues/1348. This sort behavior will likely be confusing to users. Is there a way to clarify in the UI:

- [ ] That some records may be missing values and these records will always be at the end
- [ ] That dates are the earliest_dates provided by the data provider
- [ ] Ideally we can fix the way titles sort when numbers are part of the title https://github.com/sul-dlss/dlme/issues/1192 but this is related so linking it here

Screen Shot 2022-02-16 at 2 11 55 PM

### PR DESCRIPTION
**Pull request by corylown**
Closes #1470 and #1479

### CODE REVIEW

File: app/controllers/catalog_controller.rb [hunk 1]
```diff
@@ -265,6 +265,8 @@ class CatalogController < ApplicationController
     config.add_sort_field 'relevance', sort: 'score desc, sortable_cho_title_ssi asc', label: 'Relevance'
     config.add_sort_field 'title', sort: 'sortable_cho_title_ssi asc, sortable_cho_creator_ssi asc', label: 'Title'
     config.add_sort_field 'creator', sort: 'sortable_cho_creator_ssi asc', label: 'Creator'
+    config.add_sort_field 'date_old_to_new', sort: 'cho_date_norm_min_isi asc', label: 'Date (old to new)'
```
Comment (cbeer) [inline app/controllers/catalog_controller.rb:268]:
```suggestion
    config.add_sort_field 'date_old_to_new', sort: 'cho_date_norm_min_isi asc, sortable_cho_title_ssi asc', label: 'Date (old to new)'
```

Just a thought.

File: app/controllers/session_displays_controller.rb [hunk 2]
```diff
@@ -4,9 +4,10 @@
 # A controller for disabling session information
 class SessionDisplaysController < ApplicationController
   ##
-  # Update the session, to disable_search_context
+  # Update the session, to disable_search_context or disable_date_sort_context
   def update
-    session[:disable_search_context] = true
+    session[:disable_search_context] = true if params[:context] == 'result_info'
```
Comment (cbeer) [inline app/controllers/session_displays_controller.rb:9]: Is it worth trying to align the session key with the `params[:context]` value?

File: app/views/shared/_contextual_result_info.html.erb [hunk 3]
```diff
@@ -1,4 +1,4 @@
-<div class="alert alert-info alert-dismissible fade show" role="alert">
+<div class="result_info_message alert alert-info alert-dismissible fade show" role="alert">
```
Comment (cbeer) [inline app/views/shared/_contextual_result_info.html.erb:1]: Is the `result_info_message` class used, or is it just there for our benefit? 

```suggestion
<div class="result-info-message alert alert-info alert-dismissible fade show" role="alert">
```

I think we prefer dashes in class names?

Comment (corylown) [reply app/views/shared/_contextual_result_info.html.erb:1]: I added the class here for parity between the two alerts and I wanted the class on the date sort alert to select for testing 🤷‍♂️ . I'll switch to dashes

File: app/controllers/session_displays_controller.rb (hunk 2 above)
Comment (corylown) [reply app/controllers/session_displays_controller.rb:9]: makes sense to me

File: app/controllers/catalog_controller.rb (hunk 1 above)
Comment (corylown) [reply app/controllers/catalog_controller.rb:268]: probably a good thought

Comment (cbeer) [reply app/controllers/catalog_controller.rb:268]: (and... why we didn't think that thought when we added `creator`... 🤷‍♂️ )

**Pull request merged**